ext/oauth-client/CRM/OAuth/ContactFromToken.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+class CRM_OAuth_ContactFromToken {
+
+  /**
+   * Given a token, we add a new record to civicrm_contact based on the
+   * provider's template
+   */
+  public static function createContact(array $token): array {
+    $client = \Civi\Api4\OAuthClient::get(0)
+      ->addWhere('id', '=', $token['client_id'])
+      ->execute()
+      ->single();
+    $provider = \Civi\Api4\OAuthProvider::get(0)
```

Comment:
I'd be surprised if this works at all. If it does it's a bug, because APIv4 is supposed to be strict about type checking and `checkPermissions` is a boolean not a number.
```suggestion
    $client = \Civi\Api4\OAuthClient::get(FALSE)
      ->addWhere('id', '=', $token['client_id'])
      ->execute()
      ->single();
    $provider = \Civi\Api4\OAuthProvider::get(FALSE)
```

Reply:
This could be copy-pasted  from other examples. I have tendency to put `get(0)` in this spot because it's easier to type correctly (and it's just as opaque as `get(FALSE)`.) But I think you're correct on style grounds...  `FALSE` is more consistent. (`::get(FALSE)` shows up ~150 times;  `::get(0)` shows up ~15 times.)

Functionally, they're the same (by design), though it may be surprising if you mostly think about type-hinted `class`es. Compare:

* Type-hinted classes raise an error if mismatched.
* Type-hinted scalars are automatically cast. This means that the caller/sender can supply bool-ish values, and the callee/receiver will perceive consistent bool-strict values. This is helpful given PHP's progressive approach to typing (ie you can make one function clearer without breaking all the callers).
 
Here's a little mockup of what happens when calling `setCheckPermissions(bool)`:

```php
function printme(bool $x) {
  var_export($x);
  echo "\n";
}
printme(0);
printme(false);
printme(1);
printme(true);
```
```
false
false
true
true
```

Reply:
@colemanw thanks for reviewing; I changed the params from 0 to `FALSE`.